packages/input-selection/src/LargeFirstSelection/LargeFirstInputSelector.ts

```diff
@@ -0,0 +1,409 @@
+   * @param requiredAssets Aggregate asset requirements computed from `outputs` + burn.  Used for fee/ADA expansion.
+   * @param constraints Network / wallet selection constraints (min-ADA, fee estimator, bundle size, limit…).
+   * @param changeAddressResolver Callback that assigns addresses (or further splits) for the provisional change bundles returned by the fee engine.
+   * @param implicitValue Optional implicit components (deposits, withdrawals, mint or burn)
+   * @returns An object containing Set of all inputs that will appear in the tx body, minimum fee returned by the cost model and an Array of change outputs.
+   */
+  async #expandUtxosUntilFeeCovered(
+    initialInputs: Set<Cardano.Utxo>,
+    outputs: Set<Cardano.TxOut>,
+    allAvailable: Cardano.Utxo[],
+    requiredAssets: Cardano.TokenMap,
+    constraints: SelectionConstraints,
+    changeAddressResolver: ChangeAddressResolver,
+    implicitValue?: ImplicitValue
+  ): Promise<{ finalSelection: Set<Cardano.Utxo>; fee: bigint; change: Cardano.TxOut[] }> {
+    const utxoSelectedArr = [...initialInputs];
+    const utxoRemainingArr = allAvailable
+      .filter((u) => !initialInputs.has(u))
+      .sort(([, a], [, b]) => sortByCoins(a, b)); // ADA-descending
+
+    const outputValues = toValues([...outputs]);
+
+    const changeAddress = stubMaxSizeAddress;
+
+    const implicitCoin: Required<Cardano.util.ImplicitCoin> = {
+      deposit: implicitValue?.coin?.deposit || 0n,
+      input: implicitValue?.coin?.input || 0n,
+      reclaimDeposit: implicitValue?.coin?.reclaimDeposit || 0n,
+      withdrawals: implicitValue?.coin?.withdrawals || 0n
+    };
+    const mintMap: Cardano.TokenMap = implicitValue?.mint || new Map();
+    const uniqueTxAssetIDs = uniq([...requiredAssets.keys(), ...mintMap.keys()]);
+
+    const { implicitTokensInput, implicitTokensSpend } = mintToImplicitTokens(mintMap);
+    const implicitTokens: ImplicitTokens = {
+      input: (assetId) => implicitTokensInput.get(assetId) || 0n,
+      spend: (assetId) => implicitTokensSpend.get(assetId) || 0n
+    };
+    const {
+      change,
+      fee,
+      inputs: finalInputs
+    } = await computeChangeAndAdjustForFee({
+      computeMinimumCoinQuantity: constraints.computeMinimumCoinQuantity,
+      estimateTxCosts: (utxos, changeValues) =>
+        constraints.computeMinimumCost({
+          change: changeValues.map(
+            (value) =>
+              ({
+                address: changeAddress,
+                value
+              } as Cardano.TxOut)
+          ),
+          fee: MAX_U64,
+          inputs: new Set(utxos),
+          outputs
+        }),
+      implicitValue: {
+        implicitCoin,
+        implicitTokens
+      },
+      outputValues,
+      // `DETERMINISTIC_RANDOM` (`() => 0`) is passed so that the computeChangeAndAdjustForFee
+      // always picks the *first* entry from the ADA-descending `utxoRemaining`
+      // list.
+      random: DETERMINISTIC_RANDOM,
```

Comment:
This feels a little hacky, perhaps we could refactor it slightly to decouple from random-improve internals
```suggestion
      pickAdditionalUtxo: (utxos) => utxos[0],
```

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1621/commits/15976cecb629369018a43d23d93310a99ce64c12